caom2pipe/transfer_composable.py

```diff
@@ -0,0 +1,241 @@
+#  products derived from this           promouvoir les produits dérivés
+#  software without specific prior      de ce logiciel sans autorisation
+#  written permission.                  préalable et particulière
+#                                       par écrit.
+#
+#  This file is part of the             Ce fichier fait partie du projet
+#  OpenCADC project.                    OpenCADC.
+#
+#  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+#  you can redistribute it and/or       vous pouvez le redistribuer ou le
+#  modify it under the terms of         modifier suivant les termes de
+#  the GNU Affero General Public        la “GNU Affero General Public
+#  License as published by the          License” telle que publiée
+#  Free Software Foundation,            par la Free Software Foundation
+#  either version 3 of the              : soit la version 3 de cette
+#  License, or (at your option)         licence, soit (à votre gré)
+#  any later version.                   toute version ultérieure.
+#
+#  OpenCADC is distributed in the       OpenCADC est distribué
+#  hope that it will be useful,         dans l’espoir qu’il vous
+#  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+#  without even the implied             GARANTIE : sans même la garantie
+#  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+#  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+#  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+#  General Public License for           Générale Publique GNU Affero
+#  more details.                        pour plus de détails.
+#
+#  You should have received             Vous devriez avoir reçu une
+#  a copy of the GNU Affero             copie de la Licence Générale
+#  General Public License along         Publique GNU Affero avec
+#  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+#  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+#                                       <http://www.gnu.org/licenses/>.
+#
+#  : 4 $
+#
+# ***********************************************************************
+#
+
+import logging
+import os
+
+from cadcdata import CadcDataClient
+from caom2pipe import manage_composable as mc
+
+
+__all__ = ['CadcTransfer', 'Transfer', 'VoTransfer']
+
+
+class Transfer(object):
+    """
+    No-op class to represent the actions of transferring a file from
+    one point to another.
+    """
+
+    def __init__(self):
+        self._client = None
+        self._observable = None
+
+    @property
+    def observable(self):
+        return self._observable
+
+    @observable.setter
+    def observable(self, value):
+        self._observable = value
+
+    def get(self, source, dest_fqn):
+        """
+        This no-op implementation is used when use_local_files = True.
+        :param source:
+        :param dest_fqn: str - file-system based fully-qualified name
+        """
+        pass
```

Comment:
Probably a raise NotImplemented could be used here.

Reply:
This is the default implementation, if the config has use_local_files set to True, so a 'raise' here would unexpectedly stop execution.